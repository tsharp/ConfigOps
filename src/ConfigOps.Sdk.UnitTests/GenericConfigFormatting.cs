using ConfigOps.Sdk.Api.v1alpha1;
using ConfigOps.Sdk.Serialization;
using Snapshooter.Xunit;

namespace ConfigOps.Sdk.UnitTests
{
    public class GenericConfigFormatting
    {
        [Fact]
        public void GenericConfig_Serialize()
        {
            var config = new GenericConfig
            {
                Metadata =
                {
                    Name = "global-config",
                    Namespace = "global",
                    InheritedConfigs = [
                        "global/app-config-base-1"
                    ],
                },
                Data =
                {
                    ExtensionData =
                    {
                        { "setting1", "value1" },
                        { "setting2", "value2" }
                    }
                }
            };

            config
                 .ObjectToYaml()
                 .MatchSnapshot();
        }
    }
}
