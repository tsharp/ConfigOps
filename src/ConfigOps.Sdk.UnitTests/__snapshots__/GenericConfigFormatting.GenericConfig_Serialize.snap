data:
  setting1: value1
  setting2: value2
apiVersion: config.kuipersys.com/v1alpha1
kind: GenericConfig
metadata:
  name: global-config
  namespace: global
  inheritedConfigs:
  - global/app-config-base-1
